veps/sig-storage/utility-volumes.md

```diff
@@ -0,0 +1,252 @@
+
+**Migration Behavior:**
+
+VMs with utility volumes remain **migratable** - migration is not rejected outright. When migration is initiated, it will be kept in `Pending` state as long as utility volumes are present in the VMI spec.
+> Note: Pending migrations do not count against concurrent migration slots, hence it will not block other migrations from progressing.
+
+Each operation that attached the utility volume needs to be completed and the volumes detached by their respective controllers for the migration to continue. The migration priority mechanism can influence whether operations should be aborted or wait for completion based on migration urgency.
+
+A timeout mechanism will be introduced to prevent indefinite blocking caused by a utility volume that fails to detach. A timer begins at the migration request start, and the migration must transition to `Scheduling` state before timeout expires. If not, the migration will be marked as Failed, preventing system hang and allowing retries once the underlying issue is resolved.
+
+No utility volume attachment pods are created on the destination node during migration. Utility volumes have no point in migrating as they are temporary operational tools for pod-level operations, not part of the VM's persistent state. If desired, utility volumes can be reattached after migration completion.
+
+**Rationale:** This approach balances migration responsiveness with operational continuity by allowing utility operations to complete or abort gracefully while preventing indefinite migration blocking.
+
+## API Examples
+
+### Add a utility volume
+
+#### Before:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-vmi
+spec:
+  domain:
+    # VM domain configuration
+  volumes:
+    # Existing volumes for the guest VM
+    - name: system-disk
+      containerDisk:
+        image: kubevirt/fedora-cloud-container-disk-demo:latest
+
+```
+
+#### After (controller adds a memoryDump utility volume):
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-vmi
+spec:
+  domain:
+    # remains unchanged
+  volumes:
+    # remains unchanged
+    - name: system-disk
+      containerDisk:
+        image: kubevirt/fedora-cloud-container-disk-demo:latest
+  # NEW FIELD:
+  utilityVolumes:
+    - name: memorydump-target
+      memoryDump:
+        persistentVolumeClaim:
+          claimName: memorydump-pvc
+          hotpluggable: true
+```
+
+## Alternatives
+
+Two approaches involving placing utility volumes within the existing `spec.volumes` list rather than creating a separate `utilityVolumes` field were considered:
+
+**Option A:** Add individual volume source types (e.g., `backupVolumeSource`) alongside the existing `memoryDumpVolumeSource`
+**Option B:** Create a single `utilityVolumes` volumeSource type that contains multiple utility volume types, deprecating the standalone `memoryDumpVolumeSource` field
```

Comment:
looking at the example yaml above, it looks like the utility volume contains a `memoryDumpVolumeSource` attribute. Maybe it would be useful to add the golang struct definitions for `spec.utilityVolumes`.